maze.pl

```diff
@@ -181,8 +211,14 @@
 % program that does something interesting.  Congratulations!
 %
 
+maze_solution(From,To,Path,Length) :-
+	connected(From,To),
+	connected(To,From),
+	connected(To,Path),
+	len(Length,Path).
```

Comment:
Your maze_solution should be making use of the path_to to build a path from the solution.